### PR TITLE
add case for virtio-mem access and discard setting

### PR DESCRIPTION
**Pull request by nanli1**
xxxx-299163: Virtio-mem memory device with access and discard settings
Signed-off-by: nanli <nanli@redhat.com>

rhel9 & x86 
```
avocado run --vt-type libvirt --test-runner=runner --vt-machine-type q35 memory.devices.virtio_mem.access_and_discard.with_numa
 (1/9) type_specific.io-github-autotest-libvirt.memory.devices.virtio_mem.access_and_discard.with_numa.undefined.file: PASS (83.18 s)
 (2/9) type_specific.io-github-autotest-libvirt.memory.devices.virtio_mem.access_and_discard.with_numa.undefined.anonymous: PASS (78.08 s)
 (3/9) type_specific.io-github-autotest-libvirt.memory.devices.virtio_mem.access_and_discard.with_numa.undefined.memfd: PASS (82.58 s)
 (4/9) type_specific.io-github-autotest-libvirt.memory.devices.virtio_mem.access_and_discard.with_numa.shared_and_discard.file: PASS (83.88 s)
 (5/9) type_specific.io-github-autotest-libvirt.memory.devices.virtio_mem.access_and_discard.with_numa.shared_and_discard.anonymous: PASS (81.49 s)
 (6/9) type_specific.io-github-autotest-libvirt.memory.devices.virtio_mem.access_and_discard.with_numa.shared_and_discard.memfd: PASS (78.68 s)
 (7/9) type_specific.io-github-autotest-libvirt.memory.devices.virtio_mem.access_and_discard.with_numa.private_and_no_discard.file: PASS (83.69 s)
 (8/9) type_specific.io-github-autotest-libvirt.memory.devices.virtio_mem.access_and_discard.with_numa.private_and_no_discard.anonymous: PASS (85.91 s)
 (9/9) type_specific.io-github-autotest-libvirt.memory.devices.virtio_mem.access_and_discard.with_numa.private_and_no_discard.memfd: PASS (82.90 s)

```

rhel9 & aarch64
```

avocado run --vt-type libvirt --test-runner=runner --vt-machine-type arm64-mmio   memory.devices.virtio_mem.access_and_discard
 (1/9) type_specific.io-github-autotest-libvirt.memory.devices.virtio_mem.access_and_discard.with_numa.undefined.file: PASS (70.83 s)
 (2/9) type_specific.io-github-autotest-libvirt.memory.devices.virtio_mem.access_and_discard.with_numa.undefined.anonymous: PASS (70.47 s)
 (3/9) type_specific.io-github-autotest-libvirt.memory.devices.virtio_mem.access_and_discard.with_numa.undefined.memfd: PASS (69.11 s)
 (4/9) type_specific.io-github-autotest-libvirt.memory.devices.virtio_mem.access_and_discard.with_numa.shared_and_discard.file: PASS (67.19 s)
 (5/9) type_specific.io-github-autotest-libvirt.memory.devices.virtio_mem.access_and_discard.with_numa.shared_and_discard.anonymous: PASS (70.45 s)
 (6/9) type_specific.io-github-autotest-libvirt.memory.devices.virtio_mem.access_and_discard.with_numa.shared_and_discard.memfd: PASS (69.58 s)
 (7/9) type_specific.io-github-autotest-libvirt.memory.devices.virtio_mem.access_and_discard.with_numa.private_and_no_discard.file: PASS (70.28 s)
 (8/9) type_specific.io-github-autotest-libvirt.memory.devices.virtio_mem.access_and_discard.with_numa.private_and_no_discard.anonymous: PASS (66.03 s)
 (9/9) type_specific.io-github-autotest-libvirt.memory.devices.virtio_mem.access_and_discard.with_numa.private_and_no_discard.memfd: PASS (67.12 s)



```

### CODE REVIEW

**Comment by nanli1** [comment]
@dzhengfy  Could you please help add request arm label  on this pr , and I have tested and updated it on rhel9 & aarch 64 ,PASSED

**Review by cliping — APPROVED** [review]
LGTM

File: libvirt/tests/src/memory/memory_devices/virtio_mem_access_and_discard.py [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+            single_mem.update({'mem_access': item[0]})
+        if item[1] is not None:
+            single_mem.update({'mem_discard': item[1]})
+        mem_list.append(single_mem)
+
+    test.log.debug("Get all virtio-mem list:'%s'", mem_list)
+    return mem_list
+
+
+def check_access_and_discard(test, params, vm,
```
Comment (chunfuwen) [inline libvirt/tests/src/memory/memory_devices/virtio_mem_access_and_discard.py:76]: vm , expected_share, expected_discard ,those values can be extracted from params, so they are not necessary to be pass in as independent parameters

Comment (nanli1) [reply libvirt/tests/src/memory/memory_devices/virtio_mem_access_and_discard.py:76]: Thanks you advice chunfu. There maybe sometimes  that the` def check_access_and_discard`  is not always checking the same expected result in one case. So I splitted the expected result out of the function.

**Review by chunfuwen — APPROVED** [review]
lgtm

**Pull request merged**